### DISCUSSION TITLE
UBL HOW-TO?

### DISCUSSION THREAD

**Discussion by mattwhatkins** [Jyers/Marlin#1410] [Q&A] [answered]
I'm having difficulty with UBL on my ender3-v2 with a BL touch. My glass bed is clearly higher in the middle and I had all sorts of problems with manual bed levelling with the stock firmware, hence I thought I'd try UBL. I've flashed E3V2-UBL-BLTouch-15x15-HS-v4.2.2-v1.3.5.b.bin and gone through the steps below but the printer isn't printing "level". 

1. Create New Mesh in Levelling menu
2. Autotilted current mesh
3. Adjusted some of the corners, and repeated steps 1-2 to make things a little more level. 
4. Saved the mesh

At this point, if I go into manual tuning menu, select "Go to mesh Z value", and step between the points; the distance seems good and "level" across the bed.

I'm using Creality Slicer, and have added the following commands which i've confirmed are in the resulting gcode file:

`G28 ;Home`
`G29 A ; Activate Bed Levelling`
`G29 L0 ; Load the first saved mesh`
`G29 J ; Probe the bed for mesh tilting`

EDIT:
Tried the following order as suggested by @tititopher68-dev to no avail:
`G28`
`G29 L0`
`G29 A`
`G29 J2`

But when I print, the nozzle is at the wrong height for the corners and is way off, different to that shown when I go into manual tuning. Meanwhile the centre height is correct. I'm testing with this file, the centre square adheres great, but the 4 outside squares are too high and just become a mess: https://www.thingiverse.com/thing:4642274 (My understanding is that the mesh should be used to adjust the z-offset dynamically to keep the extruder trammed against the bed?) 

Am I missing something? Should I touch the z-offset value?

**Comment by Jyers** [comment, marked as answer]
Could have been an issue caused by the marlin source. Try the latest source, a new release will be out soon
